packages/simcore-sdk/src/simcore_sdk/node_ports_common/r_clone.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+import logging
+import re
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import AsyncGenerator, Optional
+
+from aiocache import cached
+from aiofiles import tempfile
+from aiohttp import ClientSession
+from models_library.users import UserID
+from pydantic.errors import PydanticErrorMixin
+from settings_library.r_clone import RCloneSettings
+from settings_library.utils_r_clone import get_r_clone_config
+
+from .constants import ETag
+from .storage_client import (
+    delete_file_meta_data,
+    get_upload_file_presigned_link,
+    update_file_meta_data,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class _CommandFailedException(PydanticErrorMixin, RuntimeError):
+    msg_template: str = "Command {command} finished with exception:\n{stdout}"
+
+
+@asynccontextmanager
+async def _config_file(config: str) -> AsyncGenerator[str, None]:
+    async with tempfile.NamedTemporaryFile("w") as f:
+        await f.write(config)
+        await f.flush()
+        yield f.name
+
+
+async def _async_command(*cmd: str, cwd: Optional[str] = None) -> str:
+    str_cmd = " ".join(cmd)
+    proc = await asyncio.create_subprocess_shell(
+        str_cmd,
```

Comment:
you might want to use shlex.quote to prevent shell injection using this fct: see [here](https://docs.python.org/3/library/asyncio-subprocess.html#creating-subprocesses)

Reply:
Used it, but only to escape file names. Thanks for the tip, I was searching for some more advanced escaping.